app/controllers/application_controller.rb

```diff
@@ -2,4 +2,23 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def roll(min, max, bonus)
```

Comment:
Please move to a separate class DiceRoller in `app/services/dice_roller.rb`

Then create a DiceController in `app/controllers/dice_controller.rb` with a `create` action and a route to it, e.g. `resources :dice, only: [:create]`

Also missing specs 😉